src-java/flowmonitoring-topology/flowmonitoring-storm-topology/src/main/java/org/openkilda/wfm/topology/flowmonitoring/service/ActionService.java

```diff
@@ -99,12 +115,23 @@ public void processFlowLatencyMeasurement(String flowId, FlowDirection direction
         FsmKey key = getFsmKey(flowId, direction);
         FlowLatencyMonitoringFsm fsm = fsms.get(key);
         if (fsm == null) {
-            Flow flow = flowRepository.findById(flowId)
-                    .orElseThrow(() -> new IllegalStateException(format("Flow %s not found.", flowId)));
-            long maxLatency = flow.getMaxLatency() == null || flow.getMaxLatency() == 0
-                    ? Long.MAX_VALUE : flow.getMaxLatency();
-            long maxLatencyTier2 = flow.getMaxLatencyTier2() == null || flow.getMaxLatencyTier2() == 0
-                    ? Long.MAX_VALUE : flow.getMaxLatencyTier2();
+            long maxLatency;
+            long maxLatencyTier2;
+            if (flowRepository.findById(flowId).isPresent()) {
+                Flow flow = flowRepository.findById(flowId).get();
+                maxLatency = resolveMaxLatency(flow.getMaxLatency());
+                maxLatencyTier2 = resolveMaxLatency(flow.getMaxLatencyTier2());
+            } else if (haSubFlowRepository.findById(flowId).isPresent()) {
+                HaFlow haFlow = haSubFlowRepository.findById(flowId).get().getHaFlow();
+                if (haFlow != null) {
+                    maxLatency = resolveMaxLatency(haFlow.getMaxLatency());
+                    maxLatencyTier2 = resolveMaxLatency(haFlow.getMaxLatencyTier2());
+                } else {
+                    throw new IllegalStateException(format("HA-Flow for HA-Sub-Flow %s is not found.", flowId));
+                }
```

Comment:
You can invert this if, so you can avoid one nested if
if (haFlow == null) {
throw....
}
maxLatency = ...
maxLatencyTier2 = ...